tensorpipe/channel/cuda_ipc/context_impl.cc

```diff
@@ -55,7 +52,45 @@ std::shared_ptr<CudaChannel> ContextImpl::createChannel(
 }
 
 bool ContextImpl::isViable() const {
-  return foundCudaLib_;
+  if (!foundCudaLib_) {
+    TP_VLOG(4) << "Channel context " << id_
+               << " is not viable because libcuda could not be loaded";
+    return false;
+  }
+
+  int deviceCount;
+  TP_CUDA_CHECK(cudaGetDeviceCount(&deviceCount));
+  for (int i = 0; i < deviceCount; ++i) {
+    cudaDeviceProp props;
+    TP_CUDA_CHECK(cudaGetDeviceProperties(&props, i));
+
+    if (!props.unifiedAddressing) {
+      TP_VLOG(4) << "Channel context " << id_
+                 << " is not viable because CUDA device " << i
+                 << " does not have unified addressing";
+      return false;
+    }
+
+    if (!props.computeMode != cudaComputeModeDefault) {
+      TP_VLOG(4) << "Channel context " << id_
+                 << " is not viable because CUDA device " << i
+                 << " is not in default compute mode";
+      return false;
+    }
```

Comment:
Why does this matter? What happens if that's not the case?

Reply:
The other two modes are "exclusive" and "prohibited", both of which prevent the device to be accessed by an other process (cf. https://github.com/NVIDIA/cuda-samples/blob/master/Samples/simpleIPC/simpleIPC.cu#L219).

Reply:
Ok, interesting. Could you add a comment in the code with this information? Linking to that demo seems also a good idea.